chain-impl-mockchain/src/transaction/mod.rs

```diff
@@ -21,6 +21,62 @@ pub struct AuthenticatedTransaction<OutAddress, Extra> {
     pub witnesses: Vec<Witness>,
 }
 
+custom_error! {
+    #[derive(Clone, PartialEq, Eq)]
+    pub VerifyError
+        TooManyInputs {expected: usize, actual: usize }
+            = "too many inputs, expected maximum of {expected}, but received {actual}",
+        TooManyOutputs {expected: usize, actual: usize }
+            = "too many outputs, expected maximum of {expected}, but received {actual}",
+        TooManyWitnesses {expected: usize, actual: usize }
+            = "too many witnesses, expected maximum of {expected}, but received {actual}",
+        NumberOfSignaturesInvalid { expected: usize, actual: usize }
+            = "invalid number of signatures, expected {expected}, but received {actual}",
+}
+
+pub struct VerificationLimits {
+    pub max_inputs_count: usize,
+    pub max_outputs_count: usize,
+    pub max_witnesses_count: usize,
+}
```

Comment:
This is a good idea because we might want to make some restrictions configurable at the protocol level. Don't do that know, but that's a good starting point.